power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -714,52 +768,174 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
   public:
     template <class Component> using UpdateType = typename Component::UpdateType;
 
-    static bool is_update_independent(ConstDataset const& update_data) {
-        // If the batch size is (0 or) 1, then the update data for this component is 'independent'
-        if (update_data.batch_size() <= 1) {
-            return true;
-        }
+    UpdateCompIndependence check_components_independence(ConstDataset const& update_data) const {
```

Comment:
This function is REALLY complex and LONG. It is really hard to keep track of things. If its complexity can be reduced by any means, please do.

Reply:
In the upcoming refactor this should be addressed. This is an instance of my complain about daisy-chaining lambdas

Reply:
I think we should definetly scope this out to the refactor. This can be a PR in itself. But I wanted to point it out.

Reply:
As discussed, conversation resolved.